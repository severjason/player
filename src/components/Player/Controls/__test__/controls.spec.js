// @flow
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// test file
import { shallow } from 'enzyme';

import React from 'react';
import PlayerControls from '../index';

type Props = {
  playingStatus: string,
  currentSongId: number,
  minimized: boolean,
  onPlayClick: (id: number) => void,
  onForwardClick: () => void,
  onRewindClick: () => void,
};

const props: Props = {
  playingStatus: 'STOPPED',
  currentSongId: 1,
  minimized: false,
  onPlayClick: jest.fn(),
  onForwardClick: jest.fn(),
  onRewindClick: jest.fn(),
};

test ('renders correctly', () => {
  const wrapper = shallow(
    <PlayerControls {...props} />
  );
  expect(wrapper).toMatchSnapshot();
});
