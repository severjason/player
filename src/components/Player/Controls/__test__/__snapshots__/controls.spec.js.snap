// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerControls
    currentSongId={1}
    minimized={false}
    onForwardClick={[Function]}
    onPlayClick={[Function]}
    onRewindClick={[Function]}
    playingStatus="STOPPED"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="controls-element"
          onClick={[Function]}
>
          <MdFastRewind />
</div>,
        <div
          className="controls-element play"
          onClick={[Function]}
>
          <MdPlayArrow />
</div>,
        <div
          className="controls-element"
          onClick={[Function]}
>
          <MdFastForward />
</div>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MdFastRewind />,
          "className": "controls-element",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MdPlayArrow />,
          "className": "controls-element play",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MdFastForward />,
          "className": "controls-element",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="controls-element"
            onClick={[Function]}
>
            <MdFastRewind />
</div>,
          <div
            className="controls-element play"
            onClick={[Function]}
>
            <MdPlayArrow />
</div>,
          <div
            className="controls-element"
            onClick={[Function]}
>
            <MdFastForward />
</div>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MdFastRewind />,
            "className": "controls-element",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MdPlayArrow />,
            "className": "controls-element play",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MdFastForward />,
            "className": "controls-element",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
